exports/dataset.cpp

```diff
@@ -56,7 +70,24 @@ template <class T> void declare_span(py::module &m, const std::string &suffix) {
       .def("__len__", &gsl::span<T>::size)
       .def("__iter__", [](const gsl::span<T> &self) {
         return py::make_iterator(self.begin(), self.end());
-      });
+      })
+      .def("__repr__",
+       [](const gsl::span<T> &self) {
```

Comment:
You mention this being the same representation is being used elsewhere. Is it worth making the code common?